examples/js/vr/VRController.js

```diff
@@ -0,0 +1,756 @@
+
+THREE.VRController.verbosity = 0//0.5
+
+
+//  This is what makes everything so convenient. We keep track of found 
+//  controllers right here. And by adding this one update function into your
+//  animation loop we automagically update all the controller positions, 
+//  orientations, and button states. 
+
+THREE.VRController.controllers = {}
+THREE.VRController.hasGamepadEvents = 'ongamepadconnected' in window
+THREE.VRController.scanGamepads = function(){
+
+	const gamepads = navigator.getGamepads()
+	for( let i = 0; i < gamepads.length; i ++ ){
+
+		const gamepad = gamepads[ i ]
+		if( gamepad !== null && this.controllers[ gamepad.index ] === undefined ){
+
+			THREE.VRController.onGamepadConnect( gamepad )
+		}
+	}
+}
+
+
+//  These event listeners track gamepads connecting and disconnecting,
+//  create controller instances, and add them to our controllers array
+//  so that THREE.VRController.update can update them all in one go.
+
+THREE.VRController.onGamepadConnect = function( gamepad ){
+
+
+	//  Let’s create a new controller object
+	//  that’s really an extended THREE.Object3D
+	//  and pass it a reference to this gamepad.
+	
+	const
+	scope = THREE.VRController,
+	controller = new scope( gamepad )
+
+
+	//  We also need to store this reference somewhere so that we have a list
+	//  controllers that we know need updating, and by using the gamepad.index
+	//  as the key we also know which gamepads have already been found.
+
+	scope.controllers[ gamepad.index ] = controller
+
+
+	//  Let’s give the controller a little rumble; some haptic feedback to 
+	//  let the user know it’s connected and happy.
+
+	if( controller.gamepad.hapticActuators ) controller.gamepad.hapticActuators[ 0 ].pulse( 0.1, 300 )
+
+
+	//  Now we’ll broadcast a global connection event.
+	//  We’re not using THREE’s dispatchEvent because this event
+	//  is the means of delivering the controller instance.
+	//  How would we listen for events on the controller instance
+	//  if we don’t already have a reference to it?!
+
+	if( scope.verbosity >= 0.5 ) console.log( 'vr controller connected', controller )
+	window.dispatchEvent( new CustomEvent( 'vr controller connected', { detail: controller }))
+}
+window.addEventListener( 'gamepadconnected', function( event ){
+
+	THREE.VRController.onGamepadConnect( event.gamepad )
+})
+
+
+THREE.VRController.onGamepadDisconnect = function( gamepad ){
+
+
+	//  We need to find the controller that holds the reference to this gamepad.
+
+	const 
+	scope = THREE.VRController,
+	controller = scope.controllers[ gamepad.index ]
+
+
+	//  Now we can broadcast the disconnection event on the controller itself
+	//  and also delete from our controllers object. Goodbye!
+
+	if( scope.verbosity >= 0.5 ) console.log( 'vr controller disconnected', controller )
+	controller.dispatchEvent({ type: 'disconnected', controller: controller })
+	delete controller
+}
+window.addEventListener( 'gamepaddisconnected', function( event ){
```

Comment:
For that particular bit I was aiming to follow Mozilla’s guidance here: https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API/Using_the_Gamepad_API

But instead of removing only the reference to the gamepad I deleted the whole controller. But what makes you say sneaky?